kernel/common/binding.rb

```diff
@@ -54,4 +54,12 @@ def eval(expr, filename=nil, lineno=nil)
 
     Kernel.eval(expr, self, filename, lineno)
   end
+
+  def local_variables
+    eval('local_variables')
+  end
+
+  def receiver
+    @self
```

Comment:
It should indeed be `@self` here - `self` would return the `Binding` instance itself.